pkg/test/expectations/expectations.go

```diff
@@ -115,3 +117,10 @@ func ExpectCleanedUp(c client.Client) {
 		ExpectDeleted(c, &provisioner)
 	}
 }
+
+func AttemptProvisioning(c client.Client, provisioner *v1alpha1.Provisioner, pod *v1.Pod) *v1.Pod {
+	ExpectCreatedWithStatus(c, pod)
+	ExpectCreated(c, provisioner)
+	ExpectEventuallyReconciled(c, provisioner)
+	return ExpectPodExists(c, pod.GetName(), pod.GetNamespace())
```

Comment:
I like this. It looks much cleaner. I have some quick sanity checks:

- Is it possible for us to `Get` the pod before a node has been provisioned for it? If so, it's possible for this to succeed, but have the following command in tests (`ExpectNodeExists`) to fail undesirably.
- Additionally, after using this in tests, we check to see if the node exists based on the pod's nodeName. Can we ensure the pod's nodeName will exist before we finish this function?

Reply:
As long as the reconciliation loop has executed, we're guaranteed that if a node/scheduling decisions would have been made, it's made by the time the reconcile loop is complete.